data/api/server-server/content_repository.yaml

```diff
@@ -0,0 +1,293 @@
+# Copyright 2024 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+openapi: 3.1.0
+info:
+  title: Matrix Federation Content Repository API
+  version: 1.0.0
+paths:
+  "/media/download/{mediaId}":
+    get:
+      x-addedInMatrixVersion: "1.11"
+      summary: Download content from the content repository.
+      operationId: getContent
+      security:
+        - signedRequest: []
+      parameters:
+        - $ref: '#/components/parameters/mediaId'
+        - $ref: '#/components/parameters/timeout_ms'
+      responses:
+        "200":
+          description: The content that was previously uploaded.
+          headers:
+            Content-Type:
+              $ref: '#/components/headers/downloadContentType'
+          content:
+            multipart/mixed:
+              schema:
+                # This is a workaround for us not being able to say the response is required.
+                description: |-
+                  **Required.** MUST contain a `boundary` delineating exactly two
+                  parts:
+
+                  The first part has a `Content-Type` header of `application/json`
+                  and describes the media's metadata, if any. Currently, this will
+                  always be an empty object.
+
+                  The second part is either:
+
+                  1. the bytes of the media itself, using `Content-Type` and
+                     `Content-Disposition` headers as appropriate;
+                  2. or a `Location` header to redirect the caller to where the media
+                     can be retrieved. The URL at `Location` SHOULD have appropriate
+                     `Content-Type` and `Content-Disposition` headers which describe
+                     the media.
```

Comment:
I noticed that the MSC has a clause on caching. Should this be included somewhere?

> Servers SHOULD NOT cache the Location header's value as the responding server may have applied time limits on its validity. Servers which don't immediately download the media from the provided URL should re-request the media and metadata from the /download endpoint when ready for the media bytes.